### ISSUE TITLE
auth lmdbbackend: pdnsutil affected by snapshot merge

### ISSUE THREAD

**Issue by Algirdas-Z** [PowerDNS/pdns#13427]
- [x] This is not a support question, I have read [about opensource](https://www.powerdns.com/opensource.html) and will send support questions to the IRC channel, [Github Discussions](https://github.com/PowerDNS/pdns/discussions/) or the mailing list.
- [x] I have read and understood the ['out in the open' support policy](https://blog.powerdns.com/2016/01/18/open-source-support-out-in-the-open/)

 - Program: Authoritative
 - Issue type: Bug report

### Short description

pdnsutil list-zone gets blocked while lmdb snapshot merge is in progress
### Environment

 - Operating system: Rocky8
 - Software version: Auth 4.8.1 
 - Software source: PowerDNS repository

### Steps to reproduce

1. Have 2 instances synced by lightining stream
2. Change zone on instance1
3. Wait for instance1 to upload snapshot and instance2 to download it
4. On instance2 try to do pdnsutil list-zone $zone (it will hang until snapshot merge completes)

### Expected behaviour

Listing or any other read-only operation should not be blocked

### Actual behaviour

Can't list zone

**Comment by Habbie** [comment]
DNS queries did work, right?

**Comment by Algirdas-Z** [comment]
Yes, forgot to add. DNS queries works fine, this is only case with pdnsutil
